### PR TITLE
TVAULT-4678 Fix added

### PR DESCRIPTION
**Pull request by trilio-shiwanand**
https://triliodata.atlassian.net/wiki/spaces/TVO/pages/3257401345/draft+Ansible-tvault-contego+service+installation+fix+on+CentOS+RHEL

### CODE REVIEW

File: ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml [hunk 1]
```diff
@@ -16,6 +16,19 @@
 - name: Reload Daemon
   shell: systemctl daemon-reload
 
+- name: Check if centos-release-openstack-{{OPENSTACK_DIST}} is installed
+  command: rpm -q centos-release-openstack-{{OPENSTACK_DIST}}
+  register: rpm_check
+  ignore_errors: yes
```
Comment (abhijeetpatra) [inline ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml:22]: we can ignore this rpm check, instead we can check the return code of the yum update_cache task.

File: ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml [hunk 2]
```diff
@@ -33,3 +39,10 @@
         - python3-tvault-contego
      state: latest
   when: PYTHON_VERSION=="python3"
+
+- name: Removed recently installed centos-release-openstack-{{OPENSTACK_DIST}} repo
+  ansible.builtin.package:
+    name: centos-release-openstack-{{OPENSTACK_DIST | lower}}
+    state: absent
+  register: uni_pckg
```
Comment (abhijeetpatra) [inline ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml:47]: we don't need to store the result here

File: ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml [hunk 3]
```diff
@@ -16,6 +16,12 @@
 - name: Reload Daemon
   shell: systemctl daemon-reload
 
+- name: Install centos-release-openstack-{{OPENSTACK_DIST}} repo if is not installed
+  ansible.builtin.package:
```
Comment (abhijeetpatra) [inline ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml:20]: the full module name is not necessary, simple `package` should work fine

File: ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml [hunk 4]
```diff
@@ -33,3 +39,9 @@
         - python3-tvault-contego
      state: latest
   when: PYTHON_VERSION=="python3"
+
+- name: Remove recently installed openstack-{{OPENSTACK_DIST}} repo
+  package:
+    name: centos-release-openstack-{{OPENSTACK_DIST | lower}}
+    state: absent
+  when: (inst_pckg.changed == 'True')
```
[the remark this answers is not in the document]
Comment (shyam-biradar) [reply ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml:47]: Use some other result parameter like return code, check if it works in all conditions.

[the remark this answers is not in the document]
Comment (shyam-biradar) [reply ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml:47]: Got it. It's the correct approach.

**Review by shyam-biradar — APPROVED** [review]

**Pull request merged**